### ISSUE TITLE
Missing dashboard.database.widgets JSON settings

### ISSUE THREAD

**Issue by jsarnowski96** [microsoft/azuredatastudio#6368]
Issue Type: <b>Bug</b>

I am unable to add customized widgets due to lacking dashboard.database.widgets inside settings.json file. After manually inserting following JSON code:

 "dashboard.database.widgets": [
     {
         "name": "My-Widget",
         "gridItemConfig": {
             "sizex": 2,
             "sizey": 1
         },
         "widget": {
             "insights-widget": {
                 "type": {
                     "count": {
                         "dataDirection": "vertical",
                         "dataType": "number",
                         "legendPosition": "none",
                         "labelFirstColumn": false,
                         "columnsAsLabels": false
                     }
                 },
                 "queryFile": "path_ommited/activeSession.sql"
             }
         }
     }
 ]

Azure Data Studio seems to not recognize dashboard.database.widgets line at all. I tried launching ADS both as standard user and administrator, since I was suspecting that when not elevated, ADS could limit its functionalities, but it wasn't the case. I also tried inserting the code in different parts of settings.json, however to no aveil. I also accepted the monit abour preview features appearing after each launch at the bottom right corner, since I thought it might be related to this.

Let me know how can I fix this issue. Since I need this tool in my daily work in prod/prep environments, it's really urgent for me to be able to get some data feed visualized.

Best regards,
Jakub Sarnowski

Azure Data Studio version: azuredatastudio 1.8.0 (96e214593190071f0b520439929b218f9fa673de, 2019-06-06T01:59:30.203Z)
OS version: Windows_NT x64 10.0.18362

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz (8 x 1992)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|7.44GB (1.30GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (10)</summary>

Extension|Author (truncated)|Version
---|---|---
admin-pack|Mic|0.0.2
admin-tool-ext-win|Mic|0.0.1
agent|Mic|0.41.0
cms|Mic|0.3.0
dacpac|Mic|0.4.0
import|Mic|0.9.0
managed-instance-dashboard|Mic|0.3.7
powershell|ms-|2019.5.0
profiler|Mic|0.9.0
server-report|Mic|0.1.5

</details>

**Comment by jsarnowski96** [comment]
Update: At some point ADS started to recognize dashboard.database.widgets property, however after inserting code to settings.json, saving and re-connecting to the database, all changes are being removed, so only default settings.json content is in effect. Immune to launching ADS as an administrator.

**Comment by yualan** [comment]
Hi Jakub, to clarify, did you copy over the dashboard.database.widgets json block in your user settings?

**Comment by Charles-Gagnon** [comment]
@jsarnowski96 - What settings.json file are you modifying? Are you modifying the file directly or through ADS (by opening the settings UI and clicking _Edit in settings.json_ next to the one of the widget settings)?

Could you walk through the exact steps you're doing that result in the settings file being overwritten and cleared? I'm not able to repro that behavior myself so some more information is needed.

**Comment by jsarnowski96** [comment]
> 
> 
> Hi Jakub, to clarify, did you copy over the dashboard.database.widgets json block in your user settings?

Yes, I copied this property directly from Azure Data Studio widget guide and placed it at the end of settings chain.

> 
> 
> @jsarnowski96 - What settings.json file are you modifying? Are you modifying the file directly or through ADS (by opening the settings UI and clicking _Edit in settings.json_ next to the one of the widget settings)?
> 
> Could you walk through the exact steps you're doing that result in the settings file being overwritten and cleared? I'm not able to repro that behavior myself so some more information is needed.

I'm modifying it via ADS. The exact procedure is below:
1. Connect to the database X.
2. Open existing SQL query or create a new one, then connect it to the database X.
3. Execute the query and create an insight based on the query's output.
4. Copy the JSON content into dashboard.database.widget property.
5. Save settings.json
6. Disconnect from the database X.
7. Reconnect to the database X.

End result: all changes made in settings.json are being wiped out. Only the core structure of the file remains the same. Whole dashboard.database.widget property disappears.

**Comment by jsarnowski96** [comment]
Update #1 - it seems like after last update this feature started to work to some extent. My custom widgets became persistant within settings.json config file and appear on the dashboard, however some of them returns the message "Error: Query has no results to return" despite testing the query itself and getting required data from the DB. The new thing which started to happen however is that when I click on the "Chart" section of the query's results, I can only see a blank window with no generated chart, table or other kind of object. It's probably the cause of the query error displayed on the widget. I'll keep you posted.

**Comment by Charles-Gagnon** [comment]
Would you be able to include the SQL query that you're running? And did you verify using Profiler or some other tool that the query is actually being ran against the DB you're intending?

**Comment by jsarnowski96** [comment]
> 
> 
> Would you be able to include the SQL query that you're running? And did you verify using Profiler or some other tool that the query is actually being ran against the DB you're intending?

Negative, due to security policy in our company I cannot use Profiler on our production database, although like I said previously, I tested it both via ADS and SMSS and query run correctly and returned expected data. Trust me, I know what I'm doing.
